### DISCUSSION TITLE
Does it have any disadvantages to navigate without a context?

### DISCUSSION THREAD

**Discussion by escamoteur** [Milad-Akarie/auto_route_library#1998] [Q&A]
Hi,

I just found this chapter in the readme https://pub.dev/packages/auto_route#navigating-without-context
Which I (for obvious reasons ;-) very much like because it uses get_it and also because I'm always annoyed to need a context when pushing things.
Does this has any drawbacks compared to use the context?

Cheers
Thomas

**Comment by Milad-Akarie** [comment]
Hey @escamoteur  I wrote the part of the docs back when auto-scope finding wasn't supported by auto_route, the only drawback I can think of today is that accessing a certain 'Router' in the routers graph might be a challenge. 


I might add that every sub-router  widgets (Stack or Tabs) attaches a controller to the graph, the controller's life cycle is synced with the widget's, so if you obtain a sub-router and you use it after it's disposed by the corresponding widget you might get some errors.
